TypeSystem/Reflection/reflection.h

```diff
@@ -186,15 +186,16 @@ struct ReflectorImpl {
     if (!reflected_cpp_types_.count(type_index)) {
       reflected_cpp_types_.emplace(type_index, ReflectedType_Struct());
       type_reflector_(TypeSelector<T>(), Value<ReflectedType_Struct>(reflected_cpp_types_.at(type_index)));
-      reflected_type_by_type_id_.emplace(Value<ReflectedTypeBase>(reflected_cpp_types_.at(type_index)).type_id,
-                                         reflected_cpp_types_.at(type_index));
+      type_index_by_type_id_[Value<ReflectedTypeBase>(reflected_cpp_types_.at(type_index)).type_id] =
+          type_index;
     }
     return reflected_cpp_types_.at(type_index);
   }
 
   const ReflectedType& ReflectedTypeByTypeID(const TypeID type_id) {
-    if (reflected_type_by_type_id_.count(type_id)) {
-      return reflected_type_by_type_id_.at(type_id);
+    const auto cit = type_index_by_type_id_.find(type_id);
+    if (cit != type_index_by_type_id_.end()) {
+      return reflected_cpp_types_[cit->second.type_index];
```

Comment:
Maybe leave `.at()` here? IMHO it's better to throw if something goes wrong, not silently return garbage. This will also allow this function be `const`.